src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+        first_limpet = None
+        if data['relationships']['firstLimpet']['data']:
+            first_limpet = data['relationships']['firstLimpet']['data']['id']
+        rescue = Rescue(
+            uuid=UUID(data['id']),
+            client=attr['client'],
+            code_red=attr['codeRed'],
+            lang_id=attr['data']['langID'],
+            irc_nickname=attr['data']['IRCNick'],
+            board_index=attr['data']['boardIndex'],
+            mark_for_deletion=mfd,
+            platform=Platforms[attr['platform'].upper()],
+            quotes=quotes,
+            status=Status[attr['status'].upper()],
+            system=attr['system'],
+            title=attr['title'],
+            unidentified_rats=attr['unidentifiedRats'],
+            rats=rats,
+            first_limpet=first_limpet,
+            created_at=APIManager.parse_api_datetime(attr['createdAt']),
+            updated_at=APIManager.parse_api_datetime(attr['updatedAt'])
+        )
+        # rescue.outcome = attr['outcome']  # @TODO Needs a setter or something...
+        return rescue
+
+    def __init__(self, config: typing.Optional[typing.Dict]):
+        if config:
+            self.rehash_handler(config)
+
+    async def get_rescue(self, uuid: str) -> Rescue:
+        """
+        Use the API to get a rescue by UUID.
+
+        Args:
+            uuid (str): The Rescue's UUID.
+
+        Returns:
+            A Rescue object describing the Rescue found, or None if none was found.
+        """
+        if not uuid:
+            raise ValueError("uuid must be specified!")
+
+        try:
+            data = await self._call(f"rescues/{uuid}", {'include': 'rats'})
+            if data['data']:
+                return self._deserialize_rescue(data['data'], data['included'])
+        except aiohttp.ClientError:
+            return None
+
+    async def create_rescue(self, rescue: Rescue):
+        """
+        Create the case in the API.
+
+        Args:
+            rescue (Rescue): The Rescue object to create in the API.
+        """
+        data = APIManager._serialize_rescue(rescue)
+        del data['id']
+        if data['relationships']['firstLimpet']['data'] is None:
+            del data['relationships']['firstLimpet']
+        jsonapi_data = {'data': data}
+        new_rescue = await self._call('rescues', method='POST', body=jsonapi_data)
+        breakpoint()
+        return APIManager._deserialize_rescue(new_rescue)
+
+    async def update_rescue(self, rescue: Rescue):
+        """
+        Update the case in the API with the data from new_data.
+
+        Args:
+            rescue (Rescue): The Rescue object to update in the API.
+        """
+        data = APIManager._serialize_rescue(rescue)
+        jsonapi_data = {'data': data}
+        await self._call(f"rescues/{rescue.api_id}", method='PATCH', body=jsonapi_data)
+
+    # @TODO: @functools.lru_cache()
+    async def get_rat(self, uuid: str):
```

Comment:
[error] missing return type annotation
```suggestion
    async def get_rat(self, uuid: str) -> typing.Optional[Rat]:
```